ui/src/tool-panel/km-post/dialog/km-post-edit-store.ts

```diff
@@ -124,6 +180,59 @@ function validateLinkingKmPost(state: KmPostEditState): FieldValidationIssue<KmP
         } else if (!isFaithfullySaveableAsFloat(state.kmPost.gkLocationY)) {
             errors = [...errors, ...getGkCoordinateIsNotFaithfullySaveableError('gkLocationY')];
         }
+
+        const gkPoint = parseGk(
+            state.kmPost.gkSrid,
+            state.kmPost.gkLocationX,
+            state.kmPost.gkLocationY,
+        );
+        if (
+            !isWithinEastingMargin({
+                x: 0,
+                y: 0,
+                srid: state.kmPost.gkSrid ?? '',
+                ...gkPoint,
+            })
+        ) {
+            errors = [
+                ...errors,
+                {
+                    field: 'gkLocationX',
+                    reason: 'wrong-crs',
+                    type: FieldValidationIssueType.ERROR,
+                },
+                {
+                    field: 'gkLocationY',
+                    reason: 'wrong-crs',
+                    type: FieldValidationIssueType.ERROR,
+                },
+            ];
+        }
+        const pointInWgs84 = pointToWgs84({
+            x: 0,
+            y: 0,
+            srid: state.kmPost.gkSrid ?? '',
+            ...gkPoint,
+        });
+        if (
+            pointInWgs84 &&
+            (pointInWgs84.y < SOUTHERNMOST_POINT_OF_FINLAND_DEG - NORTHING_MARGIN_DEG ||
+                pointInWgs84.y > NORTHMOST_POINT_OF_FINLAND_DEG + NORTHING_MARGIN_DEG)
+        ) {
+            errors = [
+                ...errors,
+                {
+                    field: 'gkLocationX',
+                    reason: 'wrong-crs',
+                    type: FieldValidationIssueType.ERROR,
+                },
+                {
+                    field: 'gkLocationY',
+                    reason: 'wrong-crs',
+                    type: FieldValidationIssueType.ERROR,
+                },
```

Comment:
Taas molemmille kentille, samasta syystä kuin ylempänä.  Nyt kyseessä on sama virhe kuin ylempänä, mutta näitä ei ole yhdistetty sen vuoksi, että tämä on sama virhe ainoastaan koska validointi tekee tällä hetkellä kevyitä typeryyksiä jotka johtuvat (todennäköisesti) nollien käyttämisestä defaulttiarvoina. Oikeampi tapa olisi sekä tässä virheessä että ylempänä käyttää esim. kunkin GK-kaistan määriteltyä keskikohtaa defaulttiarvona, sillä 0:t antavat ihan outoja tuloksia. Tämä kuitenkin jätetään jatkokehitykseksi, sillä vaikkakin wonky, validointi on nyt kuitenkin melko käyttökelpoinen